src/script/services/tests/manifest.ts

```diff
@@ -313,3 +334,25 @@ function isStandardOrientation(orientation: string) {
   ];
   return standardOrientations.includes(orientation);
 }
+
+function iconCanLoadSuccesfully(): Promise<boolean> {
+  return new Promise((resolve) => {
+    if (mainIcon) {
+      const imageEl = new Image();
+      imageEl.src = new URL(mainIcon.src, getManifestContext().manifestUrl).toString();

```

Comment:
This will fail when cross-origin requests are blocked, right? Can you instead use this URL, which skips cross-origin:

> https://pwabuilder-safe-url.azurewebsites.net/api/getSafeUrl?checkExistsOnly=true&url={url of icon goes here}

It will return 200 if the image loads successfully.

Reply:
Latest commit uses this URL now (: . Instead of checking the response though I still do the same img. checks as img.complete + img.naturalHeight seems to be the best check I could find to cover all the potential edge cases(such as the network request returning succesfully but not with the correct data, which ends up with a blank image).